expression/builtin_cast.go

```diff
@@ -999,9 +1005,12 @@ func (b *builtinCastDecimalAsDurationSig) Clone() builtinFunc {
 }
 
 func (b *builtinCastDecimalAsDurationSig) evalDuration(row chunk.Row) (res types.Duration, isNull bool, err error) {
+	if b.tp.Decimal > types.MaxFsp {
+		return res, true, errTooBigPrecision.GenWithStackByArgs(b.tp.Decimal, "CAST", types.MaxFsp)
+	}
 	val, isNull, err := b.args[0].EvalDecimal(b.ctx, row)
 	if isNull || err != nil {
-		return res, false, err
+		return res, true, err
```

Comment:
```suggestion
		return res, isNull, err
```
is better?

Reply:
I think directly use `true` is better for understanding.